src/lib.rs

```diff
@@ -184,9 +193,9 @@ impl<I2C: embedded_hal_async::i2c::I2c> Lis2dw12<I2C> {
         let full_scale = self.full_scale_range().await?;
         let (accx, accy, accz) = self.acc().await?;
         Ok((
-            Lis2dw12::<I2C>::convert_acc_to_gs(accx, full_scale),
-            Lis2dw12::<I2C>::convert_acc_to_gs(accy, full_scale),
-            Lis2dw12::<I2C>::convert_acc_to_gs(accz, full_scale),
+            Lis2dw12::<I2C, DELAY>::convert_acc_to_gs(accx, full_scale),
```

Comment:
We should be able to replace all the `Lis2dw12::<I2C, DELAY>` with `Self`.
```suggestion
            Self::convert_acc_to_gs(accx, full_scale),
```

Reply:
Good catch, much cleaner! Done